src/components/VmActions/index.js

```diff
@@ -55,107 +91,149 @@ class VmActions extends React.Component {
     this.state = {
       removePreserveDisks: false,
     }
+
+    this.getDefaultActions = this.getDefaultActions.bind(this)
   }
 
-  render () {
-    const {
+  /**
+   * Compose main actions that appear in Card and Toolbar
+   */
+  getDefaultActions () {
+    let {
       vm,
       pool,
-      isOnCard = false,
       onStartVm,
       onStartPool,
       onShutdown,
       onRestart,
       onForceShutdown,
       onSuspend,
-      onRemove,
     } = this.props
-
     const isPool = !!pool
-    const isPoolVm = !!vm.getIn(['pool', 'id'], false) // a Pool VM is different than a Pool (definition)
-    const onStart = (isPool ? onStartPool : onStartVm)
     const status = vm.get('status')
+    const isPoolVm = !!vm.getIn(['pool', 'id'], false)
+    const onStart = (isPool ? onStartPool : onStartVm)
 
     // TODO: On the card list page, the VM's consoles would not have been fetched yet,
-    // TODO: so the tooltip on the ConsoleButton will be blank.
+    // TODO: so the tooltip on the console button will be blank.
     let consoleProtocol = ''
     if (!vm.get('consoles').isEmpty()) {
       const vConsole = vm.get('consoles').find(c => c.get('protocol') === 'spice') || vm.getIn(['consoles', 0])
       const protocol = vConsole.get('protocol').toUpperCase()
       consoleProtocol = msg.openProtocolConsole({ protocol })
     }
+
     if (vm.get('consoleInUse')) {
```

Comment:
> On the other hand, if the "in use" check is system wide, if another user has the console open and you go to open it yourself, their connection will be closed.

That's my understanding too -- it's system wide

Reply:
> This is where spinning 'consoleInUse' out to a new reducer will cause confusion. There are 2 places where consoleInUse is stored for a vm. One in vms.vmId.consoleInUse and set by SET_VM_SESSIONS. Another in consoles.vms.vmId.consoleInUse and set by SET_CONSOLE_IN_USE.

This is because of VM update every minute. Thus better save info somewhere else where it can't be deleted or rewriting.

> On the other hand, if the "in use" check is system wide, if another user has the console open and you go to open it yourself, their connection will be closed. So a prompt to you makes sense. That is a "someone else could have the console open and you'll boot them off if you open the console now" kind of check.

Not exactly, if you are not admin, you just can't download already opened console.

> What does this mean right now? At the very least, the consoles reducer should be invoked along with SET_VM_SESSIONS reducer to initialize the value from the server.

Now it's more like notification and preventing you from downloading console.